src/native/managed/cdacreader/tests/ExecutionManager/ExecutionManagerTests.cs

```diff
@@ -119,13 +119,35 @@ public void LookupNonNullOneRangeOneMethod(MockTarget.Architecture arch)
 
         var target = ExecutionManagerTestTarget.FromBuilder(emBuilder);
 
-        // test
+        // test at method start
 
         var em = target.Contracts.ExecutionManager;
         Assert.NotNull(em);
         var eeInfo = em.GetCodeBlockHandle(methodStart);
         Assert.NotNull(eeInfo);
         TargetPointer actualMethodDesc = em.GetMethodDesc(eeInfo.Value);
         Assert.Equal(expectedMethodDescAddress, actualMethodDesc);
+
+        // test middle of method
+        eeInfo = em.GetCodeBlockHandle(methodStart + 0x250);
+        Assert.NotNull(eeInfo);
+        Assert.Equal(expectedMethodDescAddress, actualMethodDesc);
+
+        // test end of method
+        eeInfo = em.GetCodeBlockHandle(methodStart + 0x450 - 1);
+        Assert.NotNull(eeInfo);
+        Assert.Equal(expectedMethodDescAddress, actualMethodDesc);
+    }
```

Comment:
The difficulty here is that the behavior changes depending on the version of the nibblemap. I talked to @davidwrighton about nibblemap behavior when the IP is outside of a managed method, and we agreed that it could be undefined.

-----------

The previous algorithm scans until it finds the previous function. If there are no previous functions in the code range it will return null.

In the new algorithm, methods may be "extended" between a few and a couple hundred bytes if there is no immediately following method. Outside of that range it will return null.

-----------

I see two ways to test what happens when trying to get a code block for a value not in a managed function.

1. Pick a value that will neither nibblemap algorithm will return a value for (ie. a code range without any functions).
2. Mock out nibblemap from the ExecutionManager tests. This is probably the preferred approach but would take more refactoring to get right.

For now, I would be in favor of option 1 and maybe switch to option 2 once we have mocking infrastructure set up.

Reply:
I added a new test which doesn't have any allocated methods and asserted we return null.